### PR TITLE
use conditions instead of criteria as workaround

### LINKED ISSUE #18269: SQL Injection in Reactive project
I don't know whether this is the right place to ask this though...

In a reactive spring project with r2dbc, I found that SQL injection is actually possible. This may happen because of me, lacking knowledge on how to use r2dbc correctly. If it is case please let me know the correct usage.

#### Setup
- Jhipster 7.7.0
- reactive with Spring WebFlux? Yes
- type of database? SQL

#### JDL
```
entity Example {
	id String
  name String
}
```

#### Change I made
- Make /api/** be accessible without authorization header
- Make GET /api/examples accept query parameter of `name`, and bind the parameter to the `findAllBy(Pageable pageable, Criteria criteria)` repository method
- Make a request to `GET /api/examples?name=foobar';DROP TABLE example;--`

#### Result
- The sql rendered was `SELECT e.id AS e_id, e.name AS e_name FROM example e WHERE name = 'foobar';DROP TABLE example;--'`
- That actually dropped the table

스크린샷 2022-04-04 오전 7 53 32

### PR DESCRIPTION
**Pull request by atomfrede**
Instead of using criteria for where clause this workaround uses conditions directly where needed (find by id).

closes #18269

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/server/templates/src/main/java/package/repository/EntityManager.java.ejs [hunk 1]
```diff
@@ -116,19 +109,19 @@ public class EntityManager {
...
      */
-    public String createSelect(SelectFromAndJoinCondition selectFrom, Class<?> entityType, Pageable pageable, Criteria criteria) {
+    public String createSelect(SelectFromAndJoinCondition selectFrom, Class<?> entityType, Pageable pageable, Condition where) {
         if (pageable != null) {
-            if (criteria != null) {
-                return createSelectImpl(selectFrom.limitOffset(pageable.getPageSize(), pageable.getOffset()).where(Conditions.just(criteria.toString())), entityType, pageable.getSort());
-            } else {
-                return createSelectImpl(selectFrom.limitOffset(pageable.getPageSize(), pageable.getOffset()), entityType, pageable.getSort());
-            }
+            return createSelectImpl(selectFrom.limitOffset(pageable.getPageSize(), pageable.getOffset()), entityType, pageable.getSort());
```
Comment (deepu105) [inline generators/server/templates/src/main/java/package/repository/EntityManager.java.ejs:114]: should the condition be passed here?

**Review by deepu105 — APPROVED** [review]

File: generators/server/templates/src/main/java/package/repository/EntityManager.java.ejs (hunk 1 above)
Comment (atomfrede) [reply generators/server/templates/src/main/java/package/repository/EntityManager.java.ejs:114]: Yes I think you're right. Hope to be able to update in during lunch.

Comment (deepu105) [reply generators/server/templates/src/main/java/package/repository/EntityManager.java.ejs:114]: I think we can merge once that is done

**Comment by deepu105** [comment]
@pascalgrimaud shall I do the release once this is merged? (using your ansible playbook)

File: generators/server/templates/src/main/java/package/repository/EntityManager.java.ejs (hunk 1 above)
Comment (deepu105) [reply generators/server/templates/src/main/java/package/repository/EntityManager.java.ejs:114]: I can also update it

**Review by deepu105 — APPROVED** [review]

**Comment by deepu105** [comment]
@atomfrede @pascalgrimaud I have updated the conditions I'll do some tests and merge once CI is green, let me know if you guys have any objections

**Comment by deepu105** [comment]
I think we are good. I checked and updated cleaned up the advisory as well so we just need to merge, release and publish

**Comment by atomfrede** [comment]
Bounty claimed https://opencollective.com/generator-jhipster/expenses/71821

**Comment by pascalgrimaud** [comment]
@atomfrede : approved

**Pull request merged**